### PR TITLE
Implementing a headless flag to disable updates and analytics

### PR DESCRIPTION
**Pull request by gregmarr**
### Purpose

Second attempt to create a headless flag without changing any existing interfaces requiring a major version increase.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [n/a] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [n/a] Snapshot of UI changes, if any.

### Reviewers

@ramramps 
@mjkkirschner 
I'm a relative newbie at C# so if there are issues with or better ways to express any of these things, please let me know.
I'm also not sure how to add a test for this functionality.
There is a Reach PR coming that uses this functionality.

### CODE REVIEW

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@
 // to distinguish one build from another. AssemblyFileVersion is specified
 // in AssemblyVersionInfo.cs so that it can be easily incremented by the
 // automated build process.
-[assembly: AssemblyVersion("1.1.1.1813")]
+[assembly: AssemblyVersion("1.1.1.1955")]
```
Comment (gregmarr) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: I think this was done automatically by Visual Studio

**Comment by gregmarr** [comment]
https://github.com/DynamoDS/Reach/pull/215

File: src/DynamoCore/Models/DynamoModel.cs [hunk 2]
```diff
@@ -416,7 +423,11 @@ public interface IStartConfiguration
             TaskProcessMode ProcessMode { get; set; }
         }
 
-      
+        public interface IStartConfiguration2 : IStartConfiguration
```
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: I'm not sure it was the right choice on my part, but for the interfaces where we have had to do this I've taken the approach of creating a new interface that describes just the new functionality, and making the concrete class implement the original interface and the new one, adding a TODO to merge them back together later, and updating the API changes doc.

will that approach work for you here?

File: src/DynamoCore/Models/DynamoModel.cs [hunk 3]
```diff
@@ -416,7 +423,11 @@ public interface IStartConfiguration
...
+        {
+            bool IsHeadless { get; set; }
```
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: can we also use this for the dynamo CLI?

File: src/DynamoCore/Models/DynamoModel.cs [hunk 4]
```diff
@@ -617,7 +632,7 @@ protected DynamoModel(IStartConfiguration config)
                     }
                     catch (Exception ex)
                     {
-                        Logger.Log(ex.Message);
```
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: seems there a lot of line endings changes that make tracking the real changes more difficult. Any idea whats going on there?

File: src/DynamoCore/Models/DynamoModel.cs (hunk 2 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: I didn't want to change any classes currently in use, for sake of versioning.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 3 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: Probably.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 4 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: Visual Studio stripping trailing whitespace.
Add w=1 to the query part of the GitHub URL to ignore whitespace changes.
This usually means just adding ?w=1, but may need to be &w=1 if there's already a ?.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 2 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: @mjkkirschner Where is the API changes doc?

Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: https://github.com/DynamoDS/Dynamo/wiki/API-Changes

seems I forgot to add some of the other changes we did as well.

**Comment by gregmarr** [comment]
@ramramps @mjkkirschner You may want to take these two PRs for 1.1.  I've been running a manual build with these changes in Fractal's dev environment for about 13 hours now, and don't have a single startup crash or random idle crash in that time.  This compares to the 64 crashes on a single server in 48 hours that we logged last week.

**Comment by ramramps** [comment]
@gregmarr You are correct. I saw the crash on our instances. 
@mjkkirschner We have to update staging today, can we get this in today? This looks good to me.

**Comment by mjkkirschner** [comment]
@gregmarr thanks for the testing, will take another look now.

File: src/DynamoCore/Models/DynamoModel.cs [hunk 5]
```diff
@@ -416,7 +423,18 @@ public interface IStartConfiguration
...
+        /// <summary>
+        /// A new interface that adds a headless flag on top of the existing
+        /// IStartConfiguration, as the existing interface can't be changed
+        /// until 2.0.0.  The flag is used to suppress update checks and
+        /// analytics.
+        /// TODO: Merge this into IStartConfiguration for 2.0.0.
+        /// </summary>
+        public interface IStartConfiguration2 : IStartConfiguration
+        {
+            bool IsHeadless { get; set; }
```
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: @gregmarr please add a summary comment to the interface member directly.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 4 above)
Comment (mjkkirschner) [inline src/DynamoCore/Models/DynamoModel.cs]: is this happening automatically or is visual studio asking you with a popup?

File: src/DynamoCore/Models/DynamoModel.cs (hunk 5 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: @mjkkirschner Done.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 4 above)
Comment (gregmarr) [inline src/DynamoCore/Models/DynamoModel.cs]: It's automatic.  I think I installed an extension to do that a long time ago.
Most of the projects I've worked on have "no trailing whitespace" as a project style.

**Comment by mjkkirschner** [comment]
@gregmarr thanks, merging this now, is the reach PR in or do we need to merge that now as well?

**Pull request merged**